packages/opentelemetry-semantic-conventions/src/stable_attributes.ts

```diff
@@ -0,0 +1,292 @@
+ * it's RECOMMENDED to:
+ * 
+ * * Use a domain-specific attribute
+ * * Set `error.type` to capture all errors, regardless of whether they are defined within the domain-specific set or not.
+ */
+export const ATTR_ERROR_TYPE = 'error.type';
+
+/**
+ * **SHOULD** be set to true if the exception event is recorded at a point where it is known that the exception is escaping the scope of the span.
+ *
+ * @note An exception is considered to have escaped (or left) the scope of a span,
+ * if that span is ended while the exception is still logically "in flight".
+ * This may be actually "in flight" in some languages (e.g. if the exception
+ * is passed to a Context manager's `__exit__` method in Python) but will
+ * usually be caught at the point of recording the exception in most languages.
+ * 
+ * It is usually not possible to determine at the point where an exception is thrown
+ * whether it will escape the scope of a span.
+ * However, it is trivial to know that an exception
+ * will escape, if one checks for an active exception just before ending the span,
+ * as done in the [example for recording span exceptions](#recording-an-exception).
+ * 
+ * It follows that an exception may still escape the scope of the span
+ * even if the `exception.escaped` attribute was not set or set to false,
+ * since the event might have been recorded at a time where it was not
+ * clear whether the exception will escape.
+ */
+export const ATTR_EXCEPTION_ESCAPED = 'exception.escaped';
+
+/**
+ * The exception message.
+ */
+export const ATTR_EXCEPTION_MESSAGE = 'exception.message';
+
+/**
+ * A stacktrace as a string in the natural representation for the language runtime. The representation is to be determined and documented by each language SIG.
+ */
+export const ATTR_EXCEPTION_STACKTRACE = 'exception.stacktrace';
+
+/**
+ * The type of the exception (its fully-qualified class name, if applicable). The dynamic type of the exception should be preferred over the static type in languages that support it.
+ */
+export const ATTR_EXCEPTION_TYPE = 'exception.type';
+
+/**
+ * HTTP request headers, `<key>` being the normalized HTTP Header name (lowercase), the value being the header values.
+ *
+ * @note Instrumentations **SHOULD** require an explicit configuration of which headers are to be captured. Including all request headers can be a security risk - explicit configuration helps avoid leaking sensitive information.
+ * The `User-Agent` header is already captured in the `user_agent.original` attribute. Users **MAY** explicitly configure instrumentations to capture them even though it is not recommended.
+ * The attribute value **MUST** consist of either multiple header values as an array of strings or a single-item array containing a possibly comma-concatenated string, depending on the way the HTTP library provides access to headers.
+ */
+export const ATTR_HTTP_REQUEST_HEADER = (key: string) => `http.request.header.${key}`;
```

Comment:
This is how I handled the template attributes. I think it should be fine but please lmk if you don't like this approach.

Reply:
Can we lowercase this with the template? `ATTR_HTTP_REQUEST_HEADER('User-Agent')` should output `http.request.header.user-agent`.
Its description says "HTTP request headers, `<key>` being the normalized HTTP Header name (lowercase), the value being the header values." [SemConv example ](https://github.com/open-telemetry/semantic-conventions/blob/01c542ccacd869ee8a329aa33c65fb2c47bf2815/docs/attributes-registry/http.md#http-attributes) also shows these lower-cased.

Reply:
FWIW, the Java semconv package does not handle lower-casing here: https://github.com/open-telemetry/semantic-conventions-java/blob/2cab896b2c95b03d0e359a34a117c235d6ebaa88/semconv/src/main/java/io/opentelemetry/semconv/AttributeKeyTemplate.java#L100-L103

I'm inclined to keep the semconv package dead simple and not have it do processing like this. 

As well, I don't think the semconv schema files have a way to tell the generating code when a particular template should have lowercased keys and when it shouldn't. For example these two `k8s.*` templates don't mention lowercasing the key: https://github.com/open-telemetry/semantic-conventions/blob/main/model/registry/k8s.yaml#L74-L85